src/ansys/simai/core/data/training_data.py

```diff
@@ -75,18 +76,34 @@ def parts(self) -> List["TrainingDataPart"]:
             for training_data_part in self.fields["parts"]
         ]
 
-    def get_subset(self, project: Identifiable["Project"]) -> Optional[str]:
+    def get_subset(self, project: Identifiable["Project"]) -> SubsetEnum:
         """Get the subset that the training data belongs to, in relation to the given project.
 
         Args:
             project: ID or :class:`model <.projects.Project>` of the project to check
                 the :class:`~.projects.Project` object for, or its ID.
 
         Returns:
-            Name of the subset that the training data belongs to in the given project.
+            SubsetEnum of the subset that the training data belongs to in the given project.
+                (e.g. <SubsetEnum.VALIDATION: 'Validation'>)
         """
         project_id = get_id_from_identifiable(project, default=self._client._current_project)
-        return self._client._api.get_training_data_subset(project_id, self.id).get("subset")
+        subset_value = self._client._api.get_training_data_subset(project_id, self.id).get("subset")
+        return SubsetEnum(subset_value)
```

Comment:
Will crash if `subset_value` is `None`, which it is for all new data until a training is launched.